### PR TITLE
CCORE766 iOS / macOS build docs

### PR DESCRIPTION
**Pull request by bdebrincunity**
### Purpose of this PR
Add docs for iOS and macOS builds

### CODE REVIEW

File: docs/ios_and_macos_builds.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+// Build the binary
+project = buildOnPlatform( 'ios', envVars, params.DEBUG )
+ipaPath = runBokkenBuild(target: "ios", projectPath: "${project}", bokkenConfig: 'com/createstudio/bokken/macos-xcode.yaml', serviceName: "story")                          
+
+// Upload the binary to AppCenter
+AppCenterURL = UploadToAppCenter(projectType: 'mac' projectPath: ipaPath, distGroups: 'QA')
+```
+
+## Bokken Usage
+We use `bokken-cli` to interact with the service. Specifically we issue commands to create the job, poll it for rediness, copy files to the VM, run shell scripts, copy artifacts back from the VM and terminate.
```
Comment (evensolberg-ffatg) [inline docs/ios_and_macos_builds.md:57]: "readiness"

File: docs/ios_and_macos_builds.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+Since Jenkins does not have the CLI tool installed, we run it in a docker container. We have a simple Dockerfile that installs the tool from a binary available on Unity's GitLab, and re-uses the image as needed. Note that there is an official docker image, but the team did such a good job minifying it that it does not have `/bin/cat`, and therefore cannot be used with Jenkins `docker.inside()` method. Using `inside()` vastly simplifies our process since it automatically mounts the workspace folder in our `bokken-cli` container. 
+
+### Example:
+```
```
Comment (evensolberg-ffatg) [inline docs/ios_and_macos_builds.md:62]:
```groovy

**Review by evensolberg-ffatg — CHANGES_REQUESTED** [review]
A couple of minor things, mainly cosmetic.

File: docs/ios_and_macos_builds.md (hunk 1 above)
Comment (bdebrincunity) [reply docs/ios_and_macos_builds.md]: 🔴 haha. Thanks!

File: docs/ios_and_macos_builds.md (hunk 2 above)
Comment (bdebrincunity) [reply docs/ios_and_macos_builds.md:61]: Do you think should include groovy as first line or remove from examples?

Comment (evensolberg-ffatg) [reply docs/ios_and_macos_builds.md:61]: Include it - it enables syntax highlighting on some platforms.

**Review by evensolberg-ffatg — APPROVED** [review]

File: docs/ios_and_macos_builds.md (hunk 2 above)
Comment (bdebrincunity) [reply docs/ios_and_macos_builds.md]: Oo the colours! Thanks.

**Review by jrowinski3d — APPROVED** [review]

**Pull request merged**